pkg/awsutils/awsutils.go

```diff
@@ -657,9 +714,15 @@ func (cache *EC2InstanceMetadataCache) attachENI(eniID string) (string, error) {
 // return ENI id, error
 func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string, subnet string) (string, error) {
 	eniDescription := eniDescriptionPrefix + cache.instanceID
+	var securityGroups []*string
+	for securityGroup := range cache.securityGroups.data {
+		//In go, the loop iterator variable is a single variable that takes different values in each loop iteration.
```

Comment:
if you just want the keys in the `data` `map[string]bool`, use the double-valued range iterator:

```go
securityGroups make([]*string, 0, len(cache.securityGroups.data))
for key, _ range cache.securityGroups.data {
    securityGroups = append(securityGroups, aws.String(key))
}
```

Also consider making the above into a `AWSStrings()` function of `StringSet` that simply returns a slice of string pointers.